### PR TITLE
[cxx-interop] Support nested templates for conditional escapability

### PR DESCRIPTION
**Pull request by Xazax-hun**
rdar://140115862

### CODE REVIEW

**Comment by Xazax-hun** [comment]
@swift-ci please smoke test

File: lib/ClangImporter/ClangImporter.cpp [hunk 1]
```diff
@@ -5080,31 +5082,41 @@ ClangTypeEscapability::evaluate(Evaluator &evaluator,
...
-        if (argEscapability == CxxEscapability::NonEscapable)
-          return CxxEscapability::NonEscapable;
+          auto argEscapability = evaluateEscapability(
+              arg.getAsType()->getUnqualifiedDesugaredType());
+          if (argEscapability == CxxEscapability::NonEscapable)
+            return CxxEscapability::NonEscapable;
+        }
+        clang::DeclContext *rec = specDecl;
+        specDecl = nullptr;
+        while ((rec = rec->getParent())) {
```
Comment (egorzhdan) [inline lib/ClangImporter/ClangImporter.cpp:5110]: Should this loop stop once `rec->isTranslationUnit()`?

**Review by egorzhdan — COMMENTED** [review]
I only have one question, otherwise LGTM

File: lib/ClangImporter/ClangImporter.cpp (hunk 1 above)
Comment (Xazax-hun) [reply lib/ClangImporter/ClangImporter.cpp:5110]: When `rec->isTranslationUnit()` is true, `rec->getParent()` returns null and the loop terminates. That being said, I just realized `rec` is a bit confusing name here.

Comment (egorzhdan) [reply lib/ClangImporter/ClangImporter.cpp:5110]: I see, that makes sense.

**Review by egorzhdan — APPROVED** [review]

**Pull request merged**